src/server/string_family.cc

```diff
@@ -597,6 +598,42 @@ void StringFamily::Get(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void StringFamily::GetDel(CmdArgList args, ConnectionContext* cntx) {
+  get_qps.Inc();
+
+  string_view key = ArgS(args, 1);
+
+  auto get_cb = [&](Transaction* t, EngineShard* shard) { return OpGet(t->GetOpArgs(shard), key); };
+
+  DVLOG(1) << "Before Get::ScheduleSingleHopT " << key;
+
+  Transaction* trans = cntx->transaction;
+  OpResult<string> result = trans->ScheduleSingleHopT(std::move(get_cb));
+
+  if (result) {
+    DVLOG(1) << "GET " << trans->DebugId() << ": " << key << " " << result.value();
+    (*cntx)->SendBulkString(*result);
+
+    auto del_cb = [](const Transaction* t, EngineShard* shard) {
```

Comment:
It's not the right way to do it because we need these operations to be atomic.
Right now it is divided into 2 separate hops and that breaks atomicity.
I do not see other option besides expanding OpGet functionality to support deletions
and implement Del inside OpGet.